torchtitan/models/norms.py

```diff
@@ -214,8 +220,16 @@ def _rms_norm_bwd_kernel_sm(
 
 
 class TritonFusedRMSNorm(torch.autograd.Function):
+    @partial(
```

Comment:
Not sure what happens if input to this wrapped function is not DTensor, but torch.Tensor, shall we only wrap it when TP is enabled?
Also this wrapping seems specific to the case when `SequenceParallel` is enabled -- which I guess if fine, because we always enable `SequenceParallel` on norm layers.

Reply:
For Non-DTensor input, we simply pass it to `func`. This requires the corresponding `placements` to be `None`. Otherwise we throw an exception.

This implementation is specific. In fact, this throws error if we call `rms_norm(x).sum().backward()` because `sum()` will make `dy`'s placement change from `[_Partial()]` into `[Replicate()]`.

Reply:
This would make DP only solution not work. I think what `local_map` should do is that for non-DTensor inputs, it should pass through instead of erroring out, and it should not create output as DTensors if inputs are not DTensors